iModelCore/ecobjects/PublicApi/ECObjects/ECSchema.h

```diff
@@ -2935,12 +2935,11 @@ struct SearchPathSchemaFileLocater : IECSchemaLocater, RefCountedBase, NonCopyab
 struct EXPORT_VTABLE_ATTRIBUTE StringSchemaLocater : IECSchemaLocater, NonCopyableClass
 {
 private:
-    bmap<bpair<SchemaKey, SchemaMatchType>, ECSchemaPtr> m_knownSchemas;
-    bmap<Utf8String, Utf8String> m_schemaStrings;
+    bmap<Utf8String, bpair<SchemaKey, Utf8String>> m_schemaStrings;

```

Comment:
Why not make it a map of SchemaKey and schema xml string?  For example ECSchemaCache uses a SchemaMap which has a SchemaKey as the key

Reply:
I suggested this approach in a call we had today because when keying on SchemaKey, we cannot use the map's find method to obtain matchtypes other than "Exact".
We'd have to iterate and check every single key, which would kill the purpose of a map. However, now I wonder how SchemaCache does it.

Reply:
To be honest I didn't check to see what SchemaMap is doing, they could have an exact match and using find_if when looking at match type.  But you can also construct the bmap with a SchemaKeyLessThan that has the match type you want to use:
```cpp
bmap<ECN::SchemaKey, ECN::ECSchemaCP, SchemaKeyLessThan<SchemaMatchType::Latest>> map;
```